File: C++/Step13-Greedy Algorithm/1541.cpp
//
//  1541.cpp
//  Step13-Greedy Algorithm
//
//  Created by 김예빈 on 2019. 9. 10..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    string str, sub_str = "";
    int sum = 0;
    bool minus = false;
    
    cin >> str;
    
    for(int i = 0; i <= str.size(); i++) {
        if(str[i] == '-' || str[i] == '+' || str[i] == '\0') {
            if(!minus)
                sum += stoi(sub_str);
            else
                sum -= stoi(sub_str);
            
            if(str[i] == '-')
                minus = true;
            
            sub_str = "";
        } else
            sub_str += str[i];
    }
    
    cout << sum;
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step15-Greedy Algorithm/1541.swift
//
//  1541.swift
//  Step15-Greedy Algorithm
//
//  Created by Yebin Kim on 2019/12/21.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

let input = readLine()!.split(separator: "-", maxSplits: 1)

let left = input[0].split(separator: "+").map{ Int($0)! }
let sum = left.reduce(0) { $0 + $1 }

if input.count < 2 {
    print(sum)
} else {
    let right = input[1].split(separator: "+").flatMap{ $0.split(separator: "-") }.map{ Int($0)! }
    let minus = right.reduce(0) { $0 + $1 }

    print(sum - minus)
}
